src/Xrefcheck/Verify.hs

```diff
@@ -511,88 +523,52 @@ verifyReference
       VerifyResult [ExternalHttpTooManyRequests retryAfter] -> retryAfter
       _ -> Nothing
 
-    isVirtual = matchesGlobPatterns root (ecIgnoreLocalRefsTo cExclusions)
+    isVirtual canonicalRoot = matchesGlobPatterns canonicalRoot (ecIgnoreLocalRefsTo cExclusions)
 
-    checkRef mAnchor referredFile = verifying $
-      unless (isVirtual referredFile) do
-        checkReferredFileIsInsideRepo referredFile
-        mFileStatus <- tryGetFileStatus referredFile
+    checkRef mAnchor referredFile canonicalRoot = verifying $
+      unless (isVirtual canonicalRoot referredFile) do
+        referredFileRelative <- case stripPrefix
+              (unCanonicalPath canonicalRoot <> [pathSeparator])
+              (unCanonicalPath referredFile) of
+                Just ps -> pure ps
+                _       -> throwError (LocalFileOutsideRepo $ unCanonicalPath referredFile)
```

Comment:
Btw, I believe that we can keep `CanonicalPath` type in `LocalFileOutsideRepo`, that would make the code clearer for a reader.

Reply:
I like the idea and the resulting code would be simpler, but I think that then we would have to hardcode the absolute paths that are reported in our CI tests. Currently, I have changed it to show the path relative to the repository root.

Reply:
Ah yes, given that we keep relative paths there now, my suggestion does not make sense anymore.